### ISSUE TITLE
Design of experiments

### ISSUE THREAD

**Issue by lurobamx** [Ultimaker/Cura#19871]
### Is your feature request related to a problem?

The design of experiments will help to optimize the printer parameters setup. If the feature would be available, it would be very easy to calibrate the printer.

### Describe the solution you'd like

Hi
I am looking for the best setting for my CR-10 S5 with a direct extruder.
This took me to perform tests keeping some parameters fix while changing one at a time (experiments) to get the right setting.
Simplify3D has the process wizard, but it the modification and naming of the individual process are manual.
The feature request is an automated design of experiments:
The user defines the parameters to consider, such as nozzle temperature, extrusión temperature, speed, etc.
The user sets the test values for each parameter individually.
The software determines the total combinations to test
The user decides the experiments to print
The user imports the model, and aligns as desired
The software copies, accommodates the models , and assignes the process defined.

The rest of the process to get the GCode done is already known.
Thanks

### Describe alternatives you've considered

Do the processes and association to models by hand. See the excel file attached with a bunch of definitions to find out the best configuration for a bridge.

### Affected users and/or printers

All users

### Additional information & file uploads

[Impresora 3D.xlsx](https://github.com/user-attachments/files/17654913/Impresora.3D.xlsx)

**Comment by GregValiant** [comment]
This is one of the things that Ultimaker does in-house for it's own printers.  I don't speak for them but I think I can say with confidence that they won't be putting time into this.  This would have to come from the community.

But...

Right now there are post processors ("Change AT Z" comes to mind) that can do much of this.  You could enable 5 instances and have each one start at a particular layer and then carry on for the rest of the file.  After the first instance the following instances would over-ride what a previous instance had done.  Each one could set a print speed, retract distance, retract speed, flow rate, etc.  The post processor "Advanced Cooling Fan Control" can be configured to change fan speeds for different features or at different layers.  "Search and Replace" can be used to replace a line like ";LAYER25" with ";LAYER:25\nM104 S210\nM220 S75\nM106 S0" which would change the print temperature, Feedrate, and fan speed at layer 25.  The next instance of search and replace might be set up to replace ";LAYER:50" with ";LAYER:50\nM104 S215\nM220 S85\nM106 S255\nM221 S95".  That could go on until you run out of layers.

The bottom line here is that much of what I see in your "experiments" can be done with the existing tools in Cura.
If you wanted to do things like adjust the line width or layer height during a print, then you would either have to splice separately sliced files together, or get fancy with support blockers configured as Infill Meshes or Cutting Meshes.
